BedrockServer.cpp

```diff
@@ -1862,3 +1909,80 @@ void BedrockServer::_acceptSockets() {
     }
 }
 
+void BedrockServer::_loadSecureData(const map<BedrockPlugin*, list<string>>& pluginSecureDataMap, const SData& args) {
+    // Do we need to load secure data? Skip loading if we don't. In this case, plugins
+    // should manage loading debug secure data inside of the plugin.
+    if (!args.isSet("-loadSecureData")) {
+        SDEBUG("Skipping loading secure data.");
+        return;
+    }
+
+    // If a scecure data host wasn't specified, use the default.
+    string secureDataHost;
+    if (!args.isSet("-secureDataHost")) {
+        secureDataHost = "localhost:4001";
+    } else {
+        secureDataHost = args["-secureDataHost"];
+    }
+
+    // Loop through the plugins that require secure data, if the file name
+    // is prefixed with socket:, load the data from a socket instead.
+    for (auto plugin : pluginSecureDataMap) {
+        const string& pluginName = SToLower(plugin.first->getName());
+
+        for (auto file : plugin.second) {
+            const string& fileName = file;
+
+            if (!SStartsWith(fileName, "socket")) {
+                // If a secure data file exists for this plugin, try and open it and read from it.
+                if (SFileExists(fileName)) {
+                    FILE* file = fopen(fileName.c_str(), "r");
+                    if (!file) {
+                        SWARN("Couldn't open file secure data file " << fileName << " for reading. Error: " << errno << ", " << strerror(errno) << ".");
+                        continue;
+                    }
+                    SINFO("Successfully opened secure data file " << fileName << " for reading.");
+
+                    // Read the whole file into a string and pass it to the plugin it belongs to.
+                    string fileContents = SFileLoad(fileName);
```

Comment:
It seems like it'd be easier for a plugin implementor to just call `SFileLoad` on a key file himself rather than implementing this API.